### PR TITLE
Graphql Dataspace API

### PR DESCRIPTION
**Pull request by abhishoya-gs**
#### What type of PR is this?
Improvement

#### What does this PR do / why is it needed ?
Introducing GraphQL APIs which take dataspace path as input and picks up mapping/runtime from defaul execution context.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java [hunk 1]
```diff
@@ -25,16 +25,38 @@ public class GraphQLProdCacheKey implements GraphQLCacheKey
     private String runtimePath;
     private String queryClassPath;
     private String query;
+    private String dataspacePath;
+    private String executionContext;
 
-    public GraphQLProdCacheKey(String groupID, String artifactId, String versionId, String mappingPath, String runtimePath, String queryClassPath, String query)
+    GraphQLProdCacheKey()
```
Comment (gs-ssh16) [inline legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java:31]: Should this be private if you intend it to be built with below static methods

File: legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java [hunk 2]
```diff
@@ -25,16 +25,38 @@ public class GraphQLProdCacheKey implements GraphQLCacheKey
...
+    {
+        GraphQLProdCacheKey graphQLProdCacheKey = new GraphQLProdCacheKey();
+        graphQLProdCacheKey.groupID = groupID;
+        graphQLProdCacheKey.artifactId = artifactId;
+        graphQLProdCacheKey.versionId = versionId;
+        graphQLProdCacheKey.dataspacePath = dataspacePath;
+        graphQLProdCacheKey.executionContext = executionContext;
+        graphQLProdCacheKey.queryClassPath = queryClassPath;
+        graphQLProdCacheKey.query = query;
+        return graphQLProdCacheKey;
```
Comment (gs-ssh16) [inline legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java:59]: Would rather prefer to have two sub classes of cache key - one for mapping, runtime and one for query, unless there is an advantage of this approach

File: legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/execute/GraphQLExecute.java [hunk 3]
```diff
@@ -232,12 +248,50 @@ private Response executeGraphQLQuery(String queryClassPath, String mappingPath,
...
+        }
+        return execute(profiles, planWithSerialized, graphQLQuery);
+    }
 
-                return Response.ok("{" +
-                        "  \"data\":" + core_external_query_graphql_transformation_transformation_introspection_query.Root_meta_external_query_graphQL_transformation_introspection_graphQLIntrospectionQuery_Class_1__Document_1__String_1_(_class, queryDoc, pureModel.getExecutionSupport()) +
-                        "}").type(MediaType.TEXT_HTML_TYPE).build();
+    private Response executeGraphQLQuery(String queryClassPath, String mappingPath, String runtimePath, Document document, GraphQLCacheKey graphQLCacheKey, MutableList<CommonProfile> profiles, Callable<PureModel> modelLoader)
+    {
+        List<SerializedNamedPlans> planWithSerialized;
```
Comment (gs-ssh16) [inline legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/execute/GraphQLExecute.java:286]: All this code is almost same as above method. As I understand, only type of cache key (or what it holds changes). Should modularize

Comment (abhishoya-gs) [reply legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/execute/GraphQLExecute.java:288]: Done

File: legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java (hunk 2 above)
Comment (abhishoya-gs) [reply legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java]: Done

File: legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java (hunk 1 above)
Comment (abhishoya-gs) [reply legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/cache/GraphQLProdCacheKey.java]: Function was removed

**Review by gs-ssh16 — APPROVED** [review]

**Pull request merged**